File: 프로그래머스/0/120880. 특이한 정렬/특이한 정렬.cs
using System;
using System.Collections.Generic;
public class Solution {
    public int[] solution(int[] numlist, int n) {
        int[] answer = new int[numlist.Length];
        List<KeyValuePair<int,int>> data = new List<KeyValuePair<int,int>>();
        
        for(int i = 0; i < numlist.Length; i++)
            data.Add(new KeyValuePair<int,int>(Math.Abs(n-numlist[i]), -numlist[i]));
            
        data.Sort((pair1, pair2) => {
        int valueComparison = pair1.Key.CompareTo(pair2.Key);
        if (valueComparison != 0)
        {
            return valueComparison;
        }
            return pair1.Value.CompareTo(pair2.Value);
        });
        
        for(int i = 0; i < data.Count; i++)
            answer[i] = Math.Abs(data[i].Value);
        
        return answer;
    }
}

File: 프로그래머스/0/120880. 특이한 정렬/특이한 정렬.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

vector<int> solution(vector<int> numlist, int n) {
    vector<pair<int,int>> _map;
    vector<int> answer;
    
    for(auto i : numlist)
        _map.push_back({abs(n-i), -i});
    
    sort(_map.begin(),_map.end());

    for(auto [distance, num] : _map)
        answer.push_back(abs(num));
    
    return answer;
}